Inputs/CurrencyInput/functions/useCurrencyInputLayout.ts

```diff
@@ -114,30 +109,31 @@ export function useCurrencyInputLayout(
       cc++
     }
 
-    return String(value).length - cc
+    return value.length - cc
   }
 
   /** Handle masking on input event */
-  function onMasking(event: Event) {
-    if (
-      currencyOptions.value!.backspace &&
-      'inputType' in event &&
-      event.inputType === 'deleteContentBackward'
-    ) {
-      return
+  function onMasking({ target }: Event) {
+    // target: HTMLInputElement
+    // @ts-expect-error Property value does not exist on type EventTarget
+    inputValue.value = masking(target.value)
+
+    if (inputValue.value) {
+      const unmaskedValue = unmasking(inputValue.value ?? '0,00')
+      prevValue.value = inputValue.value || 0
+      emits('update:modelValue', unmaskedValue)
     }
-
-    inputValue.value = masking(input.value!.value)
-    prevValue.value = inputValue.value
-
-    emits('update:modelValue', unmaskedInputValue.value)
   }
 
   /** Handle click event */
-  function onClick() {
-    const pos = getPosition(input.value!.value)
-    input.value!.focus()
-    input.value!.setSelectionRange(pos, pos)
+  function onClick({ target }: Event) {
```

Comment:
You just added 3 ts-related comments just because of this...


```
function onClick(ev: Event) { // This is probably `MouseEvent` btw
  const target = ev.target as HTMLElement

   ...
}
```